### ISSUE TITLE
0.4.x issue

### ISSUE THREAD

**Issue by orf** [swarajban/npm-cache#29]
Hey,
I've just upgraded our CI runner to use 0.4.10 of `npm-cache`, but I'm running into an issue. The first fresh `npm-cache install` runs fine, but subsequent calls that use the cache cause `ember-cli` to fail with:

```
ember -v

module.js:340
    throw err;
          ^
Error: Cannot find module 'rsvp'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object. (/Users/jkay/node_modules/ember-cli/lib/ext/promise.js:3:15)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

0.3.9 appears to work fine. Here is a list of the node_modules folder [after a full install](http://sprunge.us/fFgP) and [after a cached npm-cache install](http://sprunge.us/dOXD). A tonne of packages are missing when installed from the cache using 0.4.x, but when using 0.3.x the installation works fine and all modules are un-archived correctly.

**Comment by UncleDave** [comment]
I am also seeing this behaviour on latest `npm-cache`. It proceeds alphabetically through the modules but stops halfway through g when restoring from cache.

**Comment by Prophidys** [comment]
I got the same problem here with a big amount of modules.

This seems to be caused by this : https://github.com/swarajban/npm-cache/blob/master/cacheDependencyManagers/cacheDependencyManager.js#L92

IMO the callback just kill the disk sync or the tar process.

I tested, and this work : 

    new targz().extract(
      cachePath,
      process.cwd(),
      function onExtracted (extractErr) {
        if (extractErr) {
          error = 'error extracting ' + cachePath;
          self.cacheLogError(error);
          throw extractErr;
        }
      }
    );
    self.cacheLogInfo('done extracting');

I'm not a big expert in node so I will not try to do a PR with this patch. I read that in node you should use callback so maybe the problem is more in the tar.gz code, but on each example they give I can't see any callback on the extract function.

With this it's a little dirty because the message "done extracting" come before the process of untar is finished so you wait 3/4sec to get back your prompt.

I tried with promises : 

    targz().extract(cachePath,process.cwd())
      .then(function(){
        self.cacheLogInfo('done extracting');
      })
      .catch(function(err){
          error = 'error extracting ' + cachePath;
          self.cacheLogError(error);
      });

But it's the same problem you have to wait 3/4 sec to get back your prompt.

So I let the node expert explore this problem, for now I will use the 0.3 branch.

**Comment by swarajban** [comment]
Wow that's surprising, must have something to do with the new tar library I
added. Looking into it

**Comment by swarajban** [comment]
Ok just released 0.4.11 that uses the [Decompress](https://github.com/kevva/decompress) module to extract *tar.gz files. Thanks @Prophidys for the early investigation, made solving this much easier. 

Let me know if you have any issues
